### PR TITLE
feat(utils): Add EventEmitter to be shared between datafile-manager and optimizely-sdk

### PR DESCRIPTION
**Pull request by mjc1283**
## Summary
Currently, ProjectConfigManager and DatafileManager are both implementing their own simple event emitters. This adds an EventEmitter to the utils package that we can share.

I'll follow-up by replacing the versions in ProjectConfigManager and DatafileManager with this version

## Test plan

Unit tests

### CODE REVIEW

File: packages/utils/src/eventEmitter.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      if (listenersForEvent) {
+        const index = listenersForEvent.indexOf(listener);
+        if (index > -1) {
+          listenersForEvent.splice(index, 1);
+        }
+      }
+    }
+  }
+
+  emit(eventName: string, arg?: any) {
```
Comment (mikeng13) [inline packages/utils/src/eventEmitter.ts]: Should this be `...args` so we can pass any number of arbitrary args to the listener? I think it makes it a bit more flexible

Comment (mjc1283) [reply packages/utils/src/eventEmitter.ts]: My thought is, in the current version, you could pass an object or array if you wanted to emit multiple values. I don't have a strong preference one way or the other, but at least for the current consumers of this, they don't need to use multiple argument event listeners.

**Comment by mjc1283** [comment]
@mikeng13 The typedef will get updated automatically because we have `"declaration": true` in our tsconfig.json.

File: packages/utils/src/eventEmitter.ts (hunk 1 above)
Comment (mikeng13) [reply packages/utils/src/eventEmitter.ts]: Ok, fine by me

**Pull request closed without merging**